From efd0f9f151bf8ef1ff0108100c63750ea0d1e6d5 Mon Sep 17 00:00:00 2001
From: Vibha Yadav <yvibha@suse.com>
Date: Mon, 21 Nov 2011 16:57:23 +0530
Subject: [PATCH] Bug #651465 - event gets shifted by 5:30 hrs

Don't set the zone of event to utc by default.
---
 src/calendar/e-cal-backend-groupwise-utils.c |   10 +++++-----
 1 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/src/calendar/e-cal-backend-groupwise-utils.c b/src/calendar/e-cal-backend-groupwise-utils.c
index d671db5..5640ec4 100644
--- a/src/calendar/e-cal-backend-groupwise-utils.c
+++ b/src/calendar/e-cal-backend-groupwise-utils.c
@@ -590,7 +590,7 @@ set_rrule_from_comp (ECalComponent *comp,
 				ECalComponentDateTime *dt = (ECalComponentDateTime *) l->data;
 				if (dt->value) {
 					if (!icaltime_get_timezone (*(dt->value)))
-						icaltime_set_timezone (dt->value, default_zone ? default_zone : utc);
+						icaltime_set_timezone (dt->value, icaltimezone_get_builtin_timezone_from_tzid(dt->tzid));
 					itt_utc = icaltime_convert_to_zone (*dt->value, utc);
 					item_exdate_list = g_slist_append (item_exdate_list, icaltime_as_ical_string_r (itt_utc));
 				}
@@ -661,7 +661,7 @@ set_properties_from_cal_component (EGwItem *item,
 		e_cal_component_get_dtend (comp, &dt);
 		if (dt.value) {
 			if (!icaltime_get_timezone (*dt.value))
-				icaltime_set_timezone (dt.value, default_zone ? default_zone : utc);
+				icaltime_set_timezone (dt.value, icaltimezone_get_builtin_timezone_from_tzid(dt.tzid));
 			itt_utc = icaltime_convert_to_zone (*dt.value, utc);
 			value = icaltime_as_ical_string_r (itt_utc);
 			e_gw_item_set_end_date (item, value);
@@ -678,7 +678,7 @@ set_properties_from_cal_component (EGwItem *item,
 		e_cal_component_get_due (comp, &dt);
 		if (dt.value) {
 			if (!icaltime_get_timezone (*dt.value))
-				icaltime_set_timezone (dt.value, default_zone);
+				icaltime_set_timezone (dt.value, icaltimezone_get_builtin_timezone_from_tzid(dt.tzid));
 			itt_utc = icaltime_convert_to_zone (*dt.value, utc);
 			value = icaltime_as_ical_string_r (itt_utc);
 			e_gw_item_set_due_date (item,  value);
@@ -754,7 +754,7 @@ set_properties_from_cal_component (EGwItem *item,
 	e_cal_component_get_dtstart (comp, &dt);
 	if (dt.value) {
 		if (!icaltime_get_timezone (*dt.value))
-			icaltime_set_timezone (dt.value, default_zone);
+			icaltime_set_timezone (dt.value, icaltimezone_get_builtin_timezone_from_tzid(dt.tzid));
 		itt_utc = icaltime_convert_to_zone (*dt.value, utc);
 		value = icaltime_as_ical_string_r (itt_utc);
 		e_gw_item_set_start_date (item, value);
@@ -777,7 +777,7 @@ set_properties_from_cal_component (EGwItem *item,
 	e_cal_component_get_created (comp, &dt.value);
 	if (dt.value) {
 		if (!icaltime_get_timezone (*dt.value))
-			icaltime_set_timezone (dt.value, default_zone);
+			icaltime_set_timezone (dt.value, icaltimezone_get_builtin_timezone_from_tzid(dt.tzid));
 		itt_utc = icaltime_convert_to_zone (*dt.value, utc);
 		value = icaltime_as_ical_string_r (itt_utc);
 		e_gw_item_set_creation_date (item, value);
-- 
1.7.8

